### DISCUSSION TITLE
stable hovering drone

### DISCUSSION THREAD

**Discussion by kyoeun-space** [bitcraze/discussions#824] [Q&A] [answered]
Hello. I want to make a drone that returns to its original position even if external forces work on drone when it is hovering. This is like testing drone stability. Is it possible to make a stable hovering drone using only Crazyflie Bolt 1.1? If this is possible, what part of firmware should I modify or change?

like this : https://youtube.com/shorts/ATLzzFKgGjo?feature=share

**Comment by ataffanel** [comment, marked as answer]
Hi,

In order to achieve this king of hovering, you need some sort of positioning system. In the video you sent this is achieve using optical-flow with a down-facing camera as well as a ranging sensor to sense the height above the floor.

This can be achieved with Crazyflie using the flow deck that contains an optical flow sensor and a distance ranger.

**Reply by kyoeun-space** [reply]
There were some concerns, but they were solved. Thank you for your answer.

**Reply by ataffanel** [reply]
1. The Crazyflie/Bolt IMU will allow you to estimate the Crazyflie position within a couple of centimeter for a couple of seconds. Beyond that, the drone will drift continuously.
2. This is not possible for more than a couple of seconds

IMUs in drone are not designed for position tracking, they are not useful for anything more than covering for very brief loss of external tracking. You will need some kind of external sensing to achieve any kind of positional stability. The drone in your example is likely having a down-facing camera achieving this external tracking.
